src/dashboards/teachers/common.libsonnet

```diff
@@ -0,0 +1,111 @@
+// Video commons
+
+local grafana = import 'grafonnet/grafana.libsonnet';
+local template = grafana.template;
+local common = import '../common.libsonnet';
+
+{
+  fields: {
+    context_extensions_completion_threshold: 'context.extensions.https://w3id.org/xapi/video/extensions/completion-threshold',
+    result_extensions_length: 'result.extensions.https://w3id.org/xapi/video/extensions/length',
+    result_extensions_time: 'result.extensions.https://w3id.org/xapi/video/extensions/time',
+    verb_display_en_us: 'verb.display.en-US.keyword',
+  },
+  queries: {
+    course_key: 'context.contextActivities.parent.id.keyword:${COURSE_KEY}',
+    course_query: '(%(course_key)s OR %(school_course_session)s)' % {
+      course_key: $.queries.course_key,
+      school_course_session: $.queries.school_course_session,
+    },
+    course_session: '%(course)s:${COURSE} AND %(session)s:${SESSION}' % {
+      course: common.utils.single_escape_string(common.fields.course),
+      session: common.utils.single_escape_string(common.fields.session),
+    },
+    edx_course_key: 'SELECT `key` FROM courses_course WHERE `key`="${COURSE_KEY}"',
```

Comment:
If you prefix this query by the data source, should we apply the same pattern to other queries?

- `lrs_course_key`
- `lrs_course_query`
- ...

Reply:
I'll change it when implementing changes of the LA workshop as some dashboards will be removed.